### PR TITLE
DOC Fix EX01 issues in docstrings

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Towards #37875

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @DeaMariaLeon !

Leaving open for a bit in case others have comments, then will merge

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -1841,6 +1841,13 @@ cdef class _Period(PeriodMixin):
         Returns
         -------
         Timestamp
+
+        Examples
+        --------
+        >>> period = pd.Period('2023-1-1', freq='D')
+        >>> period = period.to_timestamp()
```
Comment (phofl) [inline pandas/_libs/tslibs/period.pyx:1848]: could we use a different name here since it is no longer a period?

Comment (DeaMariaLeon) [reply pandas/_libs/tslibs/period.pyx]: Done.. 
I also added to the DataFrame.to_timestamp example, to show the use of `freq`.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @DeaMariaLeon

**Comment by DeaMariaLeon** [comment]
Thank you

**Pull request merged**